src/main/java/ch/njol/skript/expressions/arithmetic/ExprArithmetic.java

```diff
@@ -250,10 +253,12 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 			}
 			if (returnTypes == null) { // both are object; can't determine anything
 				returnType = (Class<? extends T>) Object.class;
+				knownReturnTypes = Arithmetics.getAllReturnTypes(operator);
 			} else if (returnTypes.length == 0) { // one of the classes is known but doesn't have any operations
 				return error(firstClass, secondClass);
 			} else {
 				returnType = (Class<? extends T>) Classes.getSuperClassInfo(returnTypes).getC();
+				knownReturnTypes = Set.of(returnTypes);
```

Comment:
Does this method exist in Java 8...? IIRC it is Java 9.

Reply:
Remember this is slated for 2.9 now where we have Java 11.